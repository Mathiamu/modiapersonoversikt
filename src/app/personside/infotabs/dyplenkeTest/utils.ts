import { ReactWrapper } from 'enzyme';

export function getAktivTab(infoTabs: ReactWrapper) {
    return infoTabs.find('button[role="tab"][aria-selected=true]').text();
}

export const sakerDyplenkeTestSelectorer = {
    saksnavnOversikt: 'test-saksnavnoversikt',
    saksDokumenter: 'test-saksdokumenter',
    sakstemaListeElement: 'test-sakstemaListeElement'
};
export const utbetalingerTest = {
    oversikt: 'test-utbetalinger-oversikt',
    utbetaling: 'test-utbetaling-utbetalinglamell'
};
