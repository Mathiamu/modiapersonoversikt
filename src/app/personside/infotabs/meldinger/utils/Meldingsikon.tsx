import * as React from 'react';
import { Meldingstype } from '../../../../../models/meldinger/meldinger';
import OppmoteIkon from '../../../../../svg/OppmoteIkon';
import TelefonIkon from '../../../../../svg/TelefonIkon';
import OppgaveIkon from '../../../../../svg/OppgaveIkon';
import DokumentIkon from '../../../../../svg/DokumentIkon';
import MonologIkon from '../../../../../svg/MonologIkon';
import DialogIkon from '../../../../../svg/DialogIkon';
import styled from 'styled-components';
import { pxToRem, theme } from '../../../../../styles/personOversiktTheme';
import { UndertekstBold } from 'nav-frontend-typografi';

interface MeldingsikonProps {
    type: Meldingstype;
    erFerdigstiltUtenSvar: boolean;
    erMonolog: boolean;
    antallMeldinger: number;
}

const Styling = styled.span`
    position: relative;
    svg {
        height: ${theme.margin.px30};
        width: ${theme.margin.px30};
    }
`;

const NumberBadge = styled(UndertekstBold)`
    position: absolute;
    top: ${pxToRem(-3)};
    right: ${pxToRem(9)};
    background-color: white;
    border-radius: 50%;
    padding: 0 0.1rem;
`;

function Ikon({ props }: { props: MeldingsikonProps }) {
    switch (props.type) {
        case Meldingstype.SamtalereferatOppmøte:
            return <OppmoteIkon />;
        case Meldingstype.SamtalereferatTelefon:
            return <TelefonIkon />;
        case Meldingstype.OppgaveVarsel:
            return <OppgaveIkon />;
        case Meldingstype.DokumentVarsel:
            return <DokumentIkon />;
        default: {
            // TODO Vi må legge på et ekstra besvart / ubesvart ikon...
            if (props.erMonolog) {
                return <MonologIkon />;
            } else {
                return <DialogIkon />;
            }
        }
    }
}

function Meldingsikon(props: MeldingsikonProps) {
    return (
        <Styling>
            <Ikon props={props} />
            {props.antallMeldinger > 1 && <NumberBadge>{props.antallMeldinger}</NumberBadge>}
        </Styling>
    );
}

export default Meldingsikon;
