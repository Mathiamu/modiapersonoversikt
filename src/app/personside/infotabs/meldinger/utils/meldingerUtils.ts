import { Meldingstype, Temagruppe, Traad } from '../../../../../models/meldinger/meldinger';

export function sisteSendteMelding(traad: Traad) {
    return traad.meldinger[0];
}

export function eldsteMelding(traad: Traad) {
    return traad.meldinger[traad.meldinger.length - 1];
}

export function erMonolog(traad: Traad) {
    const bareSaksbehandler: boolean = traad.meldinger.some(melding => erMeldingFraNav(melding.meldingstype));
    const bareBruker: boolean = traad.meldinger.some(melding => erMeldingFraBruker(melding.meldingstype));

    return bareSaksbehandler !== bareBruker;
}

export function erSamtalereferat(temagruppe: Temagruppe) {
    return [
        Temagruppe.Arbeid,
        Temagruppe.Familie,
        Temagruppe.Hjelpemiddel,
        Temagruppe.Pensjon,
        Temagruppe.Øvrig,
        Temagruppe.ØkonomiskSosial,
        Temagruppe.AndreSosiale
    ].includes(temagruppe);
}

export function kanLeggesTilbake(temagruppe: Temagruppe) {
    return [
        Temagruppe.Arbeid,
        Temagruppe.Familie,
        Temagruppe.Hjelpemiddel,
        Temagruppe.Bil,
        Temagruppe.OrtopediskHjelpemiddel,
        Temagruppe.PleiepengerBarnsSykdom,
        Temagruppe.Uføretrygd,
        Temagruppe.Utland,
        Temagruppe.ØkonomiskSosial,
        Temagruppe.AndreSosiale
    ].includes(temagruppe);
}

export function erPlukkbar(temagruppe: Temagruppe) {
    return [
        Temagruppe.Arbeid,
        Temagruppe.Familie,
        Temagruppe.Hjelpemiddel,
        Temagruppe.Bil,
        Temagruppe.OrtopediskHjelpemiddel,
        Temagruppe.PleiepengerBarnsSykdom,
        Temagruppe.Uføretrygd,
        Temagruppe.Utland
    ].includes(temagruppe);
}

export function erKommunaleTjenester(temagruppe: Temagruppe) {
    return [Temagruppe.ØkonomiskSosial, Temagruppe.AndreSosiale].includes(temagruppe);
}

export function erMeldingFraBruker(meldingstype: Meldingstype) {
    return [Meldingstype.SPORSMAL_SKRIFTLIG, Meldingstype.SvarSblInngående].includes(meldingstype);
}

export function erMeldingFraNav(meldingstype: Meldingstype) {
    return [
        Meldingstype.SVAR_SKRIFTLIG,
        Meldingstype.SVAR_OPPMOTE,
        Meldingstype.SVAR_TELEFON,
        Meldingstype.SAMTALEREFERAT_TELEFON,
        Meldingstype.SAMTALEREFERAT_OPPMOTE,
        Meldingstype.SPORSMAL_MODIA_UTGAAENDE,
        Meldingstype.DOKUMENT_VARSEL,
        Meldingstype.OPPGAVE_VARSEL,
        Meldingstype.DELVIS_SVAR_SKRIFTLIG
    ].includes(meldingstype);
}

export function erMeldingVarsel(meldingstype: Meldingstype) {
    return [Meldingstype.OPPGAVE_VARSEL, Meldingstype.DOKUMENT_VARSEL].includes(meldingstype);
}

export function erMeldingSpørsmål(meldingstype: Meldingstype) {
    return [Meldingstype.SPORSMAL_MODIA_UTGAAENDE, Meldingstype.SPORSMAL_SKRIFTLIG].includes(meldingstype);
}
