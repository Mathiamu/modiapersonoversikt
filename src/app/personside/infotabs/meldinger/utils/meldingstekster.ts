import { Meldingstype, Temagruppe } from '../../../../../models/meldinger/meldinger';

export function meldingstypeTekst(meldingstype: Meldingstype) {
    switch (meldingstype) {
        case Meldingstype.SVAR_TELEFON:
            return 'Svar telefon';
        case Meldingstype.SPORSMAL_SKRIFTLIG:
            return 'Spørsmål fra bruker';
        case Meldingstype.SAMTALEREFERAT_OPPMOTE:
            return 'Samtalereferat oppmøte';
        case Meldingstype.DelvisSvarSkriftlig:
            return 'Delsvar';
        case Meldingstype.DokumentVarsel:
            return 'Dokument varsel';
        case Meldingstype.OppgaveVarsel:
            return 'OppgaveIkon varsel';
        case Meldingstype.SAMTALEREFERAT_TELEFON:
            return 'Samtalereferat telefon';
        case Meldingstype.SPORSMAL_MODIA_UTGAAENDE:
            return 'Spørsmål fra NAV';
        case Meldingstype.SVAR_OPPMOTE:
            return 'Svar oppmøte';
        case Meldingstype.SvarSblInngående:
            return 'Svar fra bruker';
        case Meldingstype.SvarSkriftlig:
            return 'Svar skriftlig';
        default:
            return 'Ukjent meldingstype';
    }
}

export function temagruppeTekst(temagruppe: Temagruppe) {
    switch (temagruppe) {
        case Temagruppe.Uføretrygd:
            return 'Uføretrygd';
        case Temagruppe.Pensjon:
            return 'Pensjon';
        case Temagruppe.Arbeid:
            return 'Arbeid';
        case Temagruppe.Utland:
            return 'Utland';
        case Temagruppe.PleiepengerBarnsSykdom:
            return 'Pleiepenger barns sykdom';
        case Temagruppe.OrtopediskHjelpemiddel:
            return 'Ortopediske hjelpemiddel';
        case Temagruppe.Bil:
            return 'Bil';
        case Temagruppe.AndreSosiale:
            return 'Andre sosiale henvendelser';
        case Temagruppe.ØkonomiskSosial:
            return 'Økonomisk sosiale henvendelser';
        case Temagruppe.Øvrig:
            return 'Øvrige henvendelser';
        case Temagruppe.Hjelpemiddel:
            return 'Hjelpemiddel';
        case Temagruppe.Familie:
            return 'Familie';
        default:
            return 'Ukjent temagruppe';
    }
}
