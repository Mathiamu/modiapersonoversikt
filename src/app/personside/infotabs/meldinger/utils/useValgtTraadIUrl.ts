import { Traad } from '../../../../../models/meldinger/meldinger';
import { hasData } from '../../../../../rest/utils/restResource';
import { useInfotabsDyplenker } from '../../dyplenker';
import { useRestResource } from '../../../../../rest/consumer/useRestResource';

export function useValgtTraadIUrl(): Traad | undefined {
    const dyplenker = useInfotabsDyplenker();
    const traaderResource = useRestResource(resources => resources.trÃ¥derOgMeldinger);
    const traader = hasData(traaderResource) ? traaderResource.data : [];
    return traader.find(dyplenker.meldinger.erValgt);
}
