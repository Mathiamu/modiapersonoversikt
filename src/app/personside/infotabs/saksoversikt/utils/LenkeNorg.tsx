import { BaseUrlsResponse } from '../../../../../models/baseurls';
import { hentBaseUrl } from '../../../../../redux/restReducers/baseurls';
import { Sakstema } from '../../../../../models/saksoversikt/sakstema';
import { sakstemakodeAlle } from '../sakstemaliste/SakstemaListe';
import { Journalpost } from '../../../../../models/saksoversikt/journalpost';
import * as React from 'react';
import { hasData, RestResource } from '../../../../../rest/utils/restResource';
import { AppState } from '../../../../../redux/reducers';
import { connect } from 'react-redux';
import { Normaltekst } from 'nav-frontend-typografi';
import { useHentPersondata } from '../../../../../utils/customHooks';
import { hasError, isPending } from '@nutgaard/use-async';

interface OwnProps {
    valgtSakstema?: Sakstema;
    geografistTilknytning?: string | null;
}

interface StateProps {
    baseUrl: string;
}

type Props = OwnProps & StateProps;

function lenkeNorg2Frontend(props: Props): string {
    const temakodeTilNorgoppslag = props.valgtSakstema ? byggSokestrengTilNorgTemaOppslag(props.valgtSakstema) : '';
    return `${props.baseUrl}/#/startsok?tema=${temakodeTilNorgoppslag}&gt=${props.geografistTilknytning}`;
}

function byggSokestrengTilNorgTemaOppslag(sakstema: Sakstema) {
    if (sakstema.temakode !== sakstemakodeAlle) {
        return sakstema.temakode;
    }
    const temaArray: string[] = sakstema.dokumentMetadata.reduce((acc: string[], dok: Journalpost) => {
        const tema = dok.temakode;
        if (acc.includes(tema)) {
            return acc;
        } else {
            return [...acc, tema];
        }
    }, []);
    return temaArray.join();
}

function hentNorg2Url(baseUrlResource: RestResource<BaseUrlsResponse>) {
    return hasData(baseUrlResource) ? hentBaseUrl(baseUrlResource.data, 'norg2-frontend') : '';
}

function LenkeNorg(props: Props) {
    const persondata = useHentPersondata();
    const geografistTilknytning =
        isPending(persondata) || hasError(persondata) ? null : persondata.data.person.geografiskTilknytning;
    const norgUrl = lenkeNorg2Frontend({ geografistTilknytning: geografistTilknytning, ...props });

    return (
        <a className="lenke" target={'_blank'} rel={'noopener noreferrer'} href={norgUrl}>
            <Normaltekst>Oversikt over enheter og tema de behandler</Normaltekst>
        </a>
    );
}

function mapStateToProps(state: AppState): StateProps {
    return {
        baseUrl: hentNorg2Url(state.restResources.baseUrl)
    };
}

export default connect(mapStateToProps)(LenkeNorg);
