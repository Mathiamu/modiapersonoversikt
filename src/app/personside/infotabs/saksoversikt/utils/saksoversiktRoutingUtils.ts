import { useMemo } from 'react';
import { useLocation } from 'react-router';
import { Sakstema } from '../../../../../models/saksoversikt/sakstema';
import { usePaths } from '../../../../routes/routing';
import { Dokument, Journalpost } from '../../../../../models/saksoversikt/journalpost';
import { useQueryParams } from '../../../../../utils/url-utils';
import { erSakerFullscreen } from './erSakerFullscreen';

interface QueryParamsForSak {
    dokument?: string;
    sakstema?: string;
}

export interface SakerRouting {
    link: (sakstema?: Sakstema, valgtDokument?: Dokument, standalone?: boolean) => string;
    route: string;
    erValgtJournalpost: (journalpost: Journalpost) => boolean;
}

export function useSakerRouting(): SakerRouting {
    const pathname = useLocation().pathname;
    const paths = usePaths();
    const queryParams = useQueryParams<QueryParamsForSak>();

    return useMemo(
        () => ({
            link: (sakstema, valgtDokumentId, standalone) =>
                `${
                    standalone || erSakerFullscreen(pathname) ? paths.sakerFullscreen : paths.saker
                }?${getSaksoversiktQueryParametere(valgtDokumentId, sakstema)}`,
            route: `${paths.saker}`,
            erValgtJournalpost: (journalpost) => inneholderValgtDokument(journalpost, queryParams.dokument)
        }),
        [paths, pathname, queryParams]
    );
}

function inneholderValgtDokument(journalpost: Journalpost, dokumentId?: string): boolean {
    return (
        !!dokumentId &&
        [
            journalpost.hoveddokument.dokumentreferanse,
            ...journalpost.vedlegg.flatMap((it) => it.dokumentreferanse)
        ].includes(dokumentId)
    );
}

function getSaksoversiktQueryParametere(dokument?: Dokument, valgtSakstema?: Sakstema) {
    const sakstemaQuery =
        valgtSakstema !== undefined && valgtSakstema.temakode !== '' ? `sakstema=${valgtSakstema.temakode}` : '';
    const dokumentQuery = dokument ? `dokument=${dokument.dokumentreferanse}` : '';
    return [sakstemaQuery, dokumentQuery].filter((it) => it).join('&');
}
