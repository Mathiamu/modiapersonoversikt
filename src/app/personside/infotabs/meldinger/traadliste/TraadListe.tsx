import * as React from 'react';
import { Traad } from '../../../../../models/meldinger/meldinger';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import styled from 'styled-components';
import theme from '../../../../../styles/personOversiktTheme';
import { sisteSendteMelding, useSokEtterMeldinger } from '../utils/meldingerUtils';
import { Input } from 'nav-frontend-skjema';
import { Normaltekst } from 'nav-frontend-typografi';
import TraadListeElement from './TraadListeElement';
import { LenkeKnapp } from '../../../../../components/common-styled-components';
import useTildelteOppgaver from '../../../../../utils/hooks/useTildelteOppgaver';
import KnappBase from 'nav-frontend-knapper';
import { useState } from 'react';
import ModalWrapper from 'nav-frontend-modal';
import BesvarFlere from './besvarflere/BesvarFlere';
import { datoSynkende } from '../../../../../utils/dateUtils';

interface Props {
    traader: Traad[];
    valgtTraad?: Traad;
    sokeord: string;
    setSokeord: (newSokeord: string) => void;
}

const PanelStyle = styled.nav`
    ${theme.hvittPanel};
    ol {
        list-style: none;
    }
`;

const SokVerktøyStyle = styled.div`
    padding: 0 ${theme.margin.layout} ${theme.margin.layout};
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;

const TraadListeStyle = styled.ol`
    > *:not(:first-child) {
        border-top: ${theme.border.skille};
    }
`;

const StyledModalWrapper = styled(ModalWrapper)`
    &.modal {
        padding: 0;
    }
`;

const InputStyle = styled.div`
    padding: ${theme.margin.layout} ${theme.margin.layout} 0;
`;

const KnappWrapperStyle = styled.div`
    display: flex;
    justify-content: flex-end;
    padding: ${theme.margin.layout} ${theme.margin.layout} 0;
`;

function TraadListe(props: Props) {
    const traaderEtterSok = useSokEtterMeldinger(props.traader, props.sokeord);

    if (props.traader.length === 0) {
        return <AlertStripeInfo>Det finnes ingen meldinger for bruker.</AlertStripeInfo>;
    }

    const meldingTekst = props.traader.length === 1 ? 'melding' : 'meldinger';

    const soketreffTekst =
        props.sokeord.length > 0
            ? `Søket traff ${traaderEtterSok.length} av ${props.traader.length} ${meldingTekst}`
            : `Totalt ${props.traader.length} ${meldingTekst}`;

    return (
        <PanelStyle>
            <SlaaSammenTraaderKnapp traader={props.traader} />
            <InputStyle>
                <Input
                    value={props.sokeord}
                    onChange={event => props.setSokeord(event.target.value)}
                    label={'Søk etter melding'}
                    className={'move-input-label'}
                />
            </InputStyle>
            <SokVerktøyStyle>
                <Normaltekst>{soketreffTekst}</Normaltekst>
                {props.sokeord !== '' && <LenkeKnapp onClick={() => props.setSokeord('')}>Alle meldinger</LenkeKnapp>}
            </SokVerktøyStyle>
            <TraadListeStyle>
                {traaderEtterSok.map(traad => (
                    <TraadListeElement
                        sokeord={props.sokeord}
                        traad={traad}
                        key={traad.traadId}
                        erValgt={traad === props.valgtTraad}
                    />
                ))}
            </TraadListeStyle>
        </PanelStyle>
    );
}

function SlaaSammenTraaderKnapp({ traader }: { traader: Traad[] }) {
    const [apen, settApen] = useState(false);
    const tildelteOppgaver = useTildelteOppgaver();

    const tildelteOppgaverIdListe = tildelteOppgaver.paaBruker.map(oppgave => oppgave.henvendelseid);

    const traaderMedTildelteOppgaver = traader.filter(traad => tildelteOppgaverIdListe.includes(traad.traadId));

    //const traaderSomHarDelsvar = traaderMedTildelteOppgaver.filter(traad => harDelsvar(traad)); TODO: Bruk denne når du er ferdig

    if (traaderMedTildelteOppgaver.length > 1) {
        return (
            <KnappWrapperStyle>
                <KnappBase type={'hoved'} onClick={() => settApen(true)}>
                    Besvar flere
                </KnappBase>
                <StyledModalWrapper contentLabel={'Besvar flere'} onRequestClose={() => settApen(false)} isOpen={apen}>
                    <BesvarFlere
                        traader={traaderMedTildelteOppgaver.sort(
                            datoSynkende(traad => sisteSendteMelding(traad).opprettetDato)
                        )}
                    />
                </StyledModalWrapper>
            </KnappWrapperStyle>
        );
    } else {
        return null;
    }
}

export default TraadListe;
