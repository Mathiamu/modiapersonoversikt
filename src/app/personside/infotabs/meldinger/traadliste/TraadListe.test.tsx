import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../test/Testprovider';
import { getTestStore } from '../../../../../test/testStore';
import { hasData } from '../../../../../rest/utils/restResource';
import TraadListe from './TraadListe';

test('Viser Traadliste', () => {
    const store = getTestStore();
    const traaderResource = getTestStore().getState().restResources.tr√•derOgMeldinger;
    const traader = hasData(traaderResource) ? traaderResource.data : [];

    const container = renderer.create(
        <TestProvider customStore={store}>
            <TraadListe
                traader={traader}
                setSkjulVarsler={() => null}
                skjulVarsler={false}
                sokeord={''}
                valgtTraad={traader[0]}
                setSokeord={() => null}
                traaderEtterSokOgFiltrering={traader}
            />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
