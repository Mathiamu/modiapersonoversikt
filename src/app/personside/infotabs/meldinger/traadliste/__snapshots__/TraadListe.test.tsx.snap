// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser Traadliste 1`] = `
.c9 {
  position: relative;
}

.c9 svg {
  height: 1.5625rem;
  width: 1.5625rem;
  opacity: 0.4;
}

.c10 {
  position: absolute;
  top: -0.1875rem;
  right: -0.5rem;
  background-color: #78706a;
  color: white;
  height: 1.1rem;
  width: 1.1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 0;
  margin-left: 0.8rem;
  overflow-wrap: break-word;
}

.c11 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c11 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c11 .order-third {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-top: 0.2rem;
}

.c13 > *:not(:last-child) {
  margin-bottom: 0.2rem;
  margin-right: 0.2rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c12 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 > *:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c14 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c6 {
  padding: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  background-color: #cce1f3;
}

.c5:focus-within {
  position: relative;
  outline: none;
  box-shadow: none;
}

.c5:focus-within::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c5 .hover-animation {
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c5:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c5:hover .hover-animation {
  -webkit-transform: translateX(0.5rem);
  -ms-transform: translateX(0.5rem);
  transform: translateX(0.5rem);
}

.c15 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c15 .nav-frontend-chevron {
  width: 1.5625rem;
}

.c0 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1),0 0 0 1px rgba(0,0,0,0.1);
}

.c0 ol {
  list-style: none;
}

.c3 {
  padding: 0 .5rem .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 > * {
  border-top: solid 0.0625rem #78706a;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c1 {
  padding: .5rem .5rem 0;
}

.c1 .skjemaelement {
  margin-bottom: 0.2rem;
}

.c1 .skjemaelement__label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c2 {
  padding: .5rem;
  margin-bottom: 0 !important;
}

<nav
  aria-label="Velg melding"
  className="c0"
>
  <article
    aria-labelledby="Helt tilfeldig ID"
  >
    <h3
      className="sr-only"
      id="Helt tilfeldig ID"
    >
      Filtrer meldinger
    </h3>
    <div
      className="c1"
    >
      <div
        className="skjemaelement move-input-label"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Søk etter melding
        </label>
        <input
          className="skjemaelement__input input--fullbredde"
          id="Helt tilfeldig ID"
          onChange={[Function]}
          placeholder="Søk etter melding"
          type="text"
          value=""
        />
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
    </div>
    <div
      className="skjemaelement skjemaelement--horisontal c2"
    >
      <input
        checked={false}
        className="skjemaelement__input checkboks"
        id="Helt tilfeldig ID"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="skjemaelement__label"
        htmlFor="Helt tilfeldig ID"
      >
        Skjul varsler
      </label>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div
      className="c3"
    >
      <p
        aria-live="assertive"
        className="typo-normal"
      >
        Totalt 1 melding
      </p>
    </div>
  </article>
  <h3
    className="sr-only"
    id="Helt tilfeldig ID"
  >
    Meldingsliste - 
    Totalt 1 melding
  </h3>
  <ol
    aria-labelledby="Helt tilfeldig ID"
    className="c4"
    tabIndex={-1}
  >
    <li
      className="c5"
    >
      <input
        checked={true}
        className="sr-only test-meldinger-melding valgt_melding  "
        id="Helt tilfeldig ID"
        name="traadliste-meldinger"
        onChange={[Function]}
        type="radio"
        value="sg838exr"
      />
      <label
        className="c6"
        htmlFor="Helt tilfeldig ID"
      >
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <span
              className="sr-only"
            >
              (
              4
              )
            </span>
            <span
              className="c9"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#3e3832"
                >
                  <path
                    d="M16.5 8.75c.476 0 .941.045 1.399.115.065-.379.101-.771.101-1.18C18 3.585 13.962.25 9 .25c-4.963 0-9 3.335-9 7.435 0 1.962.855 3.826 2.36 5.178l-1.794 3.14a.5.5 0 00.62.712l4.839-1.937c.435.124.887.218 1.348.281.022.002.065-.005.075.004.075 0 .319-.002.678-.022.679-3.514 4.233-6.291 8.374-6.291z"
                  />
                  <path
                    d="M16.5 9.75c-4.065 0-7.5 2.977-7.5 6.5 0 3.355 2.878 6.224 6.434 6.572a7.581 7.581 0 003.578-.534l3.813 1.43a.498.498 0 00.623-.691l-1.325-2.65C23.327 19.289 24 17.868 24 16.25c0-3.523-3.435-6.5-7.5-6.5z"
                  />
                </g>
              </svg>
              <p
                aria-hidden="true"
                className="typo-undertekst c10"
              >
                4
              </p>
            </span>
            <div
              className="c11"
            >
              <div
                className="order-first c12"
              >
                <p
                  className="typo-element order-second"
                >
                  Delsvar - Arbeid
                </p>
                <p
                  className="typo-normal order-first"
                >
                  27.07.2019 00:00
                </p>
              </div>
              <div
                className="order-third c13"
              >
                <div
                  className="etikett etikett--fokus"
                >
                  <span
                    className="typo-normal"
                  >
                    Under arbeid
                  </span>
                </div>
                <div
                  className="etikett etikett--info"
                >
                  <span
                    className="typo-normal"
                  >
                    Delvis besvart
                  </span>
                </div>
              </div>
              <p
                className="typo-normal order-second c14"
              >
                Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
              </p>
            </div>
          </div>
        </div>
        <div
          className="hover-animation c15"
        >
          <i
            className="nav-frontend-chevron chevronboks chevron--hoyre"
          />
        </div>
      </label>
    </li>
  </ol>
</nav>
`;
